### PR TITLE
create start.js in root path and modify the build and setup in cli

### PR DESCRIPTION
**Pull request by michael8512**
1) change the setup.js in root path, this file is only used for installing dependencies and init ./erda/config.js
2) create start.js in root path, this file is used to start the full project
3) use relative path in cli/lib/setup.js to create module_dir in .env and .erda/config.js
4) add sourceMap option in build.js

### CODE REVIEW

File: start.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+const startScheduler = () => {
+  log('Starting the scheduler');
+  cp.spawn(npmCmd, ['run', 'start'], { env: process.env, cwd: schedulerDir, stdio: 'inherit' });
+}
+
+
+// start
+const start = async ()=> {
```
Comment (McDaddy) [inline start.js:72]: remove unnecessary async?
One question. How do user know all modules are completely started？

File: cli/lib/setup.js [hunk 2]
```diff
@@ -25,29 +25,16 @@ module.exports = async (moduleName, modulePort) => {
...
-      type: 'confirm',
-      name: 'isRightPath',
-      message: `The current path is ${moduleDir}, is the right root path of project ${moduleName}?`,
-    }
-  ]);
 
-  if (!answer.isRightPath) {
+  if (path.resolve(moduleDir, `../${moduleName}`) !== moduleDir) {
+    logWarn(`the current path: '${moduleDir}' is not the root path of the project ${moduleName}`);
     process.exit(1);
```
Comment (daskyrk) [inline cli/lib/setup.js:31]: Have checked whether it is in the root directory above

File: cli/lib/build.js [hunk 3]
```diff
@@ -114,33 +112,34 @@ const checkModuleValid = ()=> {
...
+const buildAll =  async (enableSourceMap) => {
   let start = 0;
   const len = moduleList.length;
+  const buildCmd = enableSourceMap ? 'build-width-sourceMap' : 'build';
 
   while (start < len) {
     const { moduleName, moduleDir } = moduleList[start];
     logInfo(`Building ${moduleName}`);
-    await spawnSync(npmCmd, ['run', 'build'], { env: process.env, cwd: moduleDir, stdio: 'inherit' });
+    await spawnSync(npmCmd, ['run', buildCmd], { env: process.env, cwd: moduleDir, stdio: 'inherit' });
```
Comment (daskyrk) [inline cli/lib/build.js:123]: Pass `enableSourceMap` to the second argument as env variable to control build logic.So user don't have to add another 'build-width-sourceMap' script.

File: start.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+const isSetupAlready = () => {
+  [ coreDir, schedulerDir, shellDir, cliDir ].every((dir) => {
+    return fs.existsSync(join(dir, 'node_modules'));
+  });
+}
+
+if (!isSetupAlready) {
+  logError(`Please run 'node setup.js' first to init this project for basic config and dependencies`);
+  process.exit(1);
```
Comment (daskyrk) [inline start.js:43]: Prefer to ask user if auto run setup rather than just exit.

File: start.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  cp.spawn(npmCmd, ['run', 'start'], { env: process.env, cwd: schedulerDir, stdio: 'inherit' });
+}
+
+
+// start
+const start = async ()=> {
+  buildCore();
+  startCore();
+  startShell();
+  startScheduler();
```
Comment (daskyrk) [inline start.js:76]: Start all processes in one terminal is not a good idea.You have to restart all of them even just want to restart one.

**Review by daskyrk — APPROVED** [review]

**Pull request closed without merging**